src/sys/unix/mod.rs

```diff
@@ -5,6 +5,13 @@
 macro_rules! syscall {
     ($fn: ident ( $($arg: expr),* $(,)* ) ) => {{
         let res = unsafe { libc::$fn($($arg, )*) };
+        #[cfg(target_os = "hermit")]
```

Comment:
Since this code is now the same as on Unix, let's merge the two blocks.

I think it's fine to change `res == -1` to `res < 0` if that is needed on Hermit.

Reply:
I merged both block together.